src/lab1/structures/sphere/Sphere.ts

```diff
@@ -1,18 +1,31 @@
-import { Traceable } from '../../types/Traceable';
 import Vertex3D from '../vertex/Vertex3D';
 import Ray from '../ray/Ray';
 import Normal3D from '../normal/Normal';
 import { Hit } from '../../types/Hit';
+import { TraceableTransformable } from '../../../lab3/types/Transformable';
 
-export class Sphere implements Traceable {
-  public readonly center: Vertex3D;
-  public readonly radius: number;
+export class Sphere implements TraceableTransformable {
+  public center: Vertex3D;
+  public radius: number;
 
   constructor(center: Vertex3D, radius: number) {
     this.center = center;
     this.radius = radius;
   }
 
+  public translate(x: number, y: number, z: number): void {
+    this.center = this.center.translate(x, y, z);
+  }
+
+  public rotate(angleX: number, angleY: number, angleZ: number): void {
+    this.center = this.center.rotate(angleX, angleY, angleZ);
+  }
+
+  public scale(x: number, y: number, z: number): void {
+    this.center = this.center.scale(x, y, z);
```

Comment:
Scaling vertex? As far as I understood it is the save as translating this vertex. If so, this is unexpected behavior of scale method. Center should remain the same, only radius changes